rllib/offline/estimators/tests/test_ope.py

```diff
@@ -1,186 +1,204 @@
-        print(*list(mean_est.items()), sep="\n")
-        print("Stddev:")
-        print(*list(std_est.items()), sep="\n")
-        print("\n\n\n")
-
-    def test_fqe_model(self):
+        # Test OPE in DQN, during training as well as by calling evaluate()
+        results = self.algo.train()
+        ope_results = results["evaluation"]["off_policy_estimator"]
+        # Check that key exists AND is not {}
+        assert ope_results, "Did not run OPE in training!"
+        assert set(ope_results.keys()) == {
+            "is",
+            "wis",
+            "dm_fqe",
+            "dr_fqe",
+        }, "Missing keys in OPE result dict"
+
+        # Check algo.evaluate() manually as well
+        results = self.algo.train()
+        ope_results = results["evaluation"]["off_policy_estimator"]
+        assert ope_results, "Did not run OPE on call to Algorithm.evaluate()!"
+        assert set(ope_results.keys()) == {
+            "is",
+            "wis",
+            "dm_fqe",
+            "dr_fqe",
+        }, "Missing keys in OPE result dict"
+
+
+class TestFQE(unittest.TestCase):
+    """Compilation and learning tests for the Fitted-Q Evaluation model"""
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        ray.init()
+        env = CliffWalkingWallEnv()
+        cls.policy = CliffWalkingWallPolicy(
+            observation_space=env.observation_space,
+            action_space=env.action_space,
+            config={},
+        )
+        cls.gamma = 0.99
+        # Collect single episode under optimal policy
+        obs_batch = []
+        new_obs = []
+        actions = []
+        action_prob = []
+        rewards = []
+        dones = []
+        obs = env.reset()
+        done = False
+        while not done:
+            obs_batch.append(obs)
+            act, _, extra = cls.policy.compute_single_action(obs)
+            actions.append(act)
+            action_prob.append(extra["action_prob"])
+            obs, rew, done, _ = env.step(act)
+            new_obs.append(obs)
+            rewards.append(rew)
+            dones.append(done)
+        cls.batch = SampleBatch(
+            obs=obs_batch,
+            actions=actions,
+            action_prob=action_prob,
+            rewards=rewards,
+            dones=dones,
+            new_obs=new_obs,
+        )
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        ray.shutdown()
+
+    def test_fqe_compilation_and_stopping(self):
         # Test FQETorchModel for:
         # (1) Check that it does not modify the underlying batch during training
-        # (2) Check that the stoppign criteria from FQE are working correctly
+        # (2) Check that the stopping criteria from FQE are working correctly
         # (3) Check that using fqe._compute_action_probs equals brute force
         # iterating over all actions with policy.compute_log_likelihoods
         fqe = FQETorchModel(
-            policy=self.algo.get_policy(),
+            policy=self.policy,
```

Comment:
I just realized the delta parameter. Can we change the name to something like `stop_loss_threshold` something that is more informative?

Reply:
And also for tau: let's use `polyak_coef`?